stream/build.gradle.kts

```
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("com.hedera.block.protobuf")
+    alias(libs.plugins.pbj)
+}
+
+group = "com.hedera.block"
+
+description = "Hedera API"
+
+// Remove the following line to enable all 'javac' lint checks that we have turned on by default
+// and then fix the reported issues.
+tasks.withType<JavaCompile>().configureEach {
+    options.compilerArgs.add("-Xlint:-exports,-deprecation,-removal")
+}
+
+// Add downloaded HAPI repo protobuf files into build directory and add to sources to build them
+tasks.cloneHederaProtobufs {
+    // uncomment below to use a specific tag
+    // tag = "v0.53.0"
+    // uncomment below to use a specific branch
+    branch = "main"
+}
+
+sourceSets {
+    main {
+        pbj {
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("services") })
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("block") })
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("platform") })
+            srcDir(tasks.cloneHederaProtobufs.flatMap { it.localCloneDirectory.dir("streams") })
+        }
```

```suggestion
        }
    }
    test {
```

**(?)** IIUC, this is what you want? Use the "protoc compiled hapi" only in testing and have no dependencies on them in production code? I tried this change, and discovered that the "protoc" code is actually used in production (server/src/main). You can do this change and then you see what no longer compiles.

---

protoc is required in production temporarily because Helidon uses the protoc types for gRPC stubs.  Once PBJ can generate stubs for Helidon (that work is in progress) we can move the protoc types to test-only.

---

Thanks for explaining. 👍